airflow/providers/google/cloud/transfers/gcs_to_sftp.py

```diff
@@ -129,15 +134,25 @@ def execute(self, context):
                 )
 
             prefix, delimiter = self.source_object.split(WILDCARD, 1)
+            prefix_dirname = os.path.dirname(prefix)
+
             objects = gcs_hook.list(self.source_bucket, prefix=prefix, delimiter=delimiter)
 
             for source_object in objects:
-                destination_path = os.path.join(self.destination_path, source_object)
+                if self.ignore_gcs_path_in_destination:
+                    source_object_basename = os.path.relpath(source_object, start=prefix_dirname)
+                    destination_path = os.path.join(self.destination_path, source_object_basename)
+                else:
+                    destination_path = os.path.join(self.destination_path, source_object)
                 self._copy_single_object(gcs_hook, sftp_hook, source_object, destination_path)
 
             self.log.info("Done. Uploaded '%d' files to %s", len(objects), self.destination_path)
         else:
-            destination_path = os.path.join(self.destination_path, self.source_object)
+            if self.ignore_gcs_path_in_destination:
+                source_object_basename = os.path.basename(self.source_object)
+                destination_path = os.path.join(self.destination_path, source_object_basename)
+            else:
+                destination_path = os.path.join(self.destination_path, self.source_object)
```

Comment:
How about
```python
def _resolve_destination_path(self, prefix=None):
    if self.ignore_gcs_path_in_destination:
        source_object_basename = os.path.basename(self.source_object, start=prefix)
        return os.path.join(self.destination_path, source_object_basename)
    else:
        return os.path.join(self.destination_path, self.source_object)
```
then we can avoid duplication of the code

Reply:
Thanks @turbaszek , it is a great suggestion.
I modified it slightly: https://github.com/apache/airflow/pull/11284/commits/9b9ef602a170125344c3b03c1eaca5c077719eb5